packages/xdl/src/Web.js

```diff
@@ -79,39 +121,158 @@ export async function onlySupportsWebAsync(projectRoot) {
   return false;
 }
 
-export async function ensureWebSupportAsync(projectRoot) {
+async function promptToAddWebPlatform() {
+  // clearConsole();
+  const question = {
+    type: 'confirm',
+    name: 'should',
+    message: `It appears you don't explicitly define "${chalk.underline(
+      `web`
+    )}" as one of the supported "${chalk.underline(`platforms`)}" in your project's ${chalk.white(
+      `app.json`
+    )}. \n  Would you like to add it?`,
+    default: true,
+  };
+  const answer = await inquirer.prompt(question);
+  return !!answer.should;
+}
+
+async function addWebPlatformToAppConfig(projectRoot) {
+  const { exp } = await readConfigJsonAsync(projectRoot);
+
+  let currentPlatforms = [];
+  if (Array.isArray(exp.platforms) && exp.platforms.length) {
+    currentPlatforms = exp.platforms;
+  }
+  if (currentPlatforms.includes('web')) {
+    return;
+  }
+
+  await ConfigUtils.writeConfigJsonAsync(projectRoot, { platforms: [...currentPlatforms, 'web'] });
+}
+
+function isUsingYarn(projectRoot) {
+  const yarnLockPath = path.resolve(projectRoot, 'yarn.lock');
+  return fs.existsSync(yarnLockPath);
+}
+
+async function installAsync(projectRoot, libraries = []) {
+  const options = {
+    cwd: projectRoot,
+    stdio: 'inherit',
+  };
+  try {
+    if (await isUsingYarn(projectRoot)) {
+      await spawnAsync('yarn', ['add', ...libraries], options);
+    } else {
+      await spawnAsync('npm', ['install', '--save', ...libraries], options);
+    }
+  } catch (error) {
+    throw new XDLError(
+      XDLError.WEB_NOT_CONFIGURED,
+      'Failed to install libraries: ' + error.message
+    );
+  }
+}
+
+function getModulesPath(projectRoot) {
+  const workspaceRoot = findWorkspaceRoot(projectRoot); // Absolute path or null
+  return path.resolve(workspaceRoot || projectRoot, 'node_modules');
+}
+
+function areModulesInstalled(projectRoot) {
+  const modulesPath = getModulesPath(projectRoot);
+  return fs.existsSync(modulesPath);
+}
+
+async function attemptToGetPackageVersionAtPathAsync(modulesPath, moduleName) {
+  const possibleModulePath = path.resolve(modulesPath, moduleName, 'package.json');
```

Comment:
You can use ``ConfigUtils.resolveModule(`${moduleName}/package.json`, projectRoot, exp)``, which takes in account custom paths for node_modules (can be overridden in app.json) and resolves the packages using the Node module resolution algorithm (like `require.resolve`) so it should also work in cases like a project that's using Yarn workspaces etc.

Reply:
Can I use `ConfigUtils.resolveModule(``, projectRoot, exp);` to determine if the `node_modules` have been installed?